src/floris/tools/floris_interface.py

```diff
@@ -1067,40 +1294,76 @@ def calculate_AEP_wind_limit(self, num_turbines, x_spacing, start_ws, threshold)
         self.max_power = self.get_farm_power()
         self.ws_limit = ws
 
-    def change_turbine(self, turb_num_array, turbine_change_dict, update_specified_wind_height=False):
+    def copy_and_update_turbine_map(
+        self, base_turbine_id: str, update_parameters: dict, new_id: str | None = None
+    ) -> dict:
+        """Creates a new copy of an existing turbine and updates the parameters based on
+        user input. This function is a helper to make the v2 -> v3 transition easier.
+
+        Args:
+            base_turbine_id (str): The base turbine's ID in `floris.farm.turbine_id`.
+            update_parameters (dict): A dictionary of the turbine parameters to update
+                and their new valies.
+            new_id (str, optional): The new `turbine_id`, if `None` a unique
+                identifier will be appended to the end. Defaults to None.
+
+        Returns:
+            dict: A turbine mapping that can be passed directly to `change_turbine`.
+        """
+        if new_id is None:
+            new_id = f"{base_turbine_id}_copy{self.unique_copy_id}"
+            self.unique_copy_id += 1
+
+        turbine = {new_id: self.floris.turbine[base_turbine_id]._asdict()}
+        turbine[new_id].update(update_parameters)
+        return turbine
+
+    def change_turbine(
+        self,
+        turbine_indices: list[int],
+        new_turbine_map: dict[str, dict[str, Any]],
```

Comment:
Is this new_turbine_map a map for all the turbines in the farm?

Reply:
No, this would be as many or as few that need to get added to be updating for new turbines. That said, we're sticking with a single turbine model, so I'm not sure this will end up being used.